lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -353,6 +353,32 @@ FileUnit *SerializedModuleLoader::loadAST(
   return nullptr;
 }
 
+bool SerializedModuleLoader::canImportModule(std::pair<Identifier, SourceLoc> moduleID) {
+  std::unique_ptr<llvm::MemoryBuffer> moduleInputBuffer;
+  std::unique_ptr<llvm::MemoryBuffer> moduleDocInputBuffer;
+  // First see if we find it in the registered memory buffers.
+  if (!MemoryBuffers.empty()) {
+    // FIXME: Right now this works only with access paths of length 1.
+    // Once submodules are designed, this needs to support suffix
+    // matching and a search path.
+    auto bufIter = MemoryBuffers.find(moduleID.first.str());
+    if (bufIter != MemoryBuffers.end()) {
+      moduleInputBuffer = std::move(bufIter->second);
+      MemoryBuffers.erase(bufIter);
+    }
+  }
+
+  // Otherwise look on disk.
+  bool isFramework = false;
+  if (!moduleInputBuffer) {
+    if (!findModule(Ctx, moduleID, moduleInputBuffer, moduleDocInputBuffer,
+                    isFramework)) {
+      return false;
+    }
+  }
+  return (moduleInputBuffer != nullptr);
```

Comment:
It's probably worth avoiding the construction of the llvm::MemoryBuffers. Maybe we should be passing pointers instead of references?